rust/experimental/server/src/lib.rs

```diff
@@ -75,8 +81,21 @@ pub async fn start_uniffle_worker(config: config::Config) -> Result<()> {
         let service = ShuffleServerServer::new(shuffle_server)
             .max_decoding_message_size(usize::MAX)
             .max_encoding_message_size(usize::MAX);
-        let _ = Server::builder().add_service(service).serve(addr).await;
+        let _ = Server::builder().add_service(service).serve_with_shutdown(addr, async {
+            rx.await.expect("graceful_shutdown fail");
+            println!("Successfully received the shutdown signal.");
+        }).await;
     });
+
+    tokio::spawn(async move {
```

Comment:
The concrete runtime has been wrapped in the runtime manager. Please respect this.

Reply:
I see default_runtime has default_runtime, it's for not important tasks, how about use it?